site/content/kbld/docs/latest/config.md

```diff
@@ -4,85 +4,294 @@ title: Configuration
+- `minimumRequiredVersion` (optional; semver) — specifies the minimum required version of `kbld` needed to work with this configuration.
+- `searchRules` — a list of [Search Rules](#search-rules) that `kbld` follows to locate container image references within the input files.
+- `overrides` — a list of [Overrides](#overrides) that `kbld` applies to found container image references before it attempts to resolve or build the actual image.
+- `sources` — a list of [Sources](#sources), each of which describes where to find the source that makes up the contents of a given image _and_ which build tool to use to construct it.
+- `destinations` — a list of [Destinations](#destinations), each of which describes where (i.e. to which container registry) to publish a given image.
 
-sources:
-- image: adservice
-  path: src/
+_(Note: prior to v0.28.0, different types of configuration were supplied in different `kind` of files: `kind: Sources`, `kind: ImageDestinations`, `kind: ImageOverrides`, `kind: ImageKeys`. Since v0.28.0, all such configuration can be specified in one kind of file: `kind: Config`; this  is recommended.)_
 
-destinations:
-- image: adservice
-  newImage: docker.io/dkalinin/microservices-demo-adservice
+---
+## Search Rules
+
+`kbld` scans input files for references to container images. Search rules describe how `kbld` should identify those references and how to process them.
+
+A search rule has two parts:
+- conditions for matching (either by key, by value, or both), and
+- a strategy for how to parse and update a matched item.
 
+A search rule is expressed like this:
+
+```yaml
+---
+apiVersion: kbld.k14s.io/v1alpha1
+kind: Config
+searchRules:
+- keyMatcher: ...
+  valueMatcher: ...
+  updateStrategy: ...
+```
+
+where a rule consists of one or both of the following...
+- `keyMatcher` a [Key Matcher](#key-matcher) — identifies container image references based on an item's key.
+- `valueMatcher` a [Value Matcher](#value-matcher) — identifies container image references based on an item's value.
+ 
+_(If both are specified, their results are "and'ed" together.)_
+
+...and optionally,
+- `updateStrategy` an [Update Strategy](#update-strategy) — what `kbld` should do with the matched container image reference.
+
+**Multiple Matching search rules** \
+If multiple search rules match the same item, the rule that was defined first is applied.
+
+Note: `kbld` includes its own [Default Search Rules](#default-search-rules)
+
+### Key Matcher
+
+Specifies whether a given key indicates that its corresponding value is an image reference.
+
+```yaml
+---
+apiVersion: kbld.k14s.io/v1alpha1
+kind: Config
 searchRules:
 - keyMatcher:
-    name: sidecarImage
+    name:
+    path:
+```
+
+where:
+- `name` (string) specifies the key name (e.g. `sidecarImage`)
+- `path` (array) specifies key path from the root of the YAML document.\
+  Each path part can be one of:
+  - the literal key name. (e.g. `[data, sidecarImage]` maps to `/data/sidecarImage`)
```

Comment:
What  does`/data/sidecarImage` mean here? I thought that `[data, sidecarImage]` would map to the path from the root of the document `data.sidecarImage`, based on the current documentation.

Reply:
You are right, ma'am! Not sure why I shifted in JSON Path notation.... will fix.